hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -204,22 +286,83 @@ def generate_novel_prefix(self, random):
         assert not self.is_exhausted
         novel_prefix = bytearray()
 
-        def append_int(n_bits, value):
-            novel_prefix.extend(int_to_bytes(value, bits_to_bytes(n_bits)))
+        BUFFER_SIZE = 8 * 1024
+
+        def draw(ir_type, kwargs, *, forced=None):
+            cd = ConjectureData(max_length=BUFFER_SIZE, prefix=b"", random=random)
+            draw_func = getattr(cd, f"draw_{ir_type}")
+            value = draw_func(**kwargs, forced=forced)
+            # using floats as keys into branch.children breaks things, because
+            # e.g. hash(0.0) == hash(-0.0) would collide as keys when they are
+            # in fact distinct child branches.
+            # To distinguish floats here we'll use their bits representation. This
+            # entails some bookkeeping such that we're careful about when the
+            # float key is in its bits form (as a key into branch.children) and
+            # when it is in its float form (as a value we want to write to the
+            # buffer), and converting between the two forms as appropriate.
+            # TODO write a test for this to confirm my intuition of breakage is
+            # correct.
+            if ir_type == "float":
+                value = float_to_int(value)
+            return (value, cd.buffer)
```

Comment:
This also breaks database playback, I think. I haven't looked into why yet. (there's a relevant failing test on this branch somewhere.)

Reply:
this might be fine, actually? I was basing this off a local failure in manual testing, but I the remaining database failures seem unrelated. 

I'll leave this comment unresolved as a TODO for myself to double check before merge, but I expect this to be fine.